packages/frontend/src/components/common/Table/TableProvider.tsx

```diff
@@ -49,28 +49,22 @@ const rowColumn: TableColumn = {
 };
 
 export const TableProvider: FC<Props> = ({ children, ...rest }) => {
-    const { data, columns, columnOrder, pagination, onColumnOrderChange } =
-        rest;
+    const { data, columns, columnOrder, pagination } = rest;
     const [columnVisibility, setColumnVisibility] = React.useState({});
-    const allColumnIds = columns.reduce<string[]>(
-        (acc, col) => (col.id ? [...acc, col.id] : acc),
-        [],
-    );
-    const reactTableColumnOrder = [
-        ...new Set([
+    const [tempColumnOrder, setTempColumnOrder] =
+        React.useState<ColumnOrderState>([
             ROW_NUMBER_COLUMN_ID,
             ...(columnOrder || []),
-            ...allColumnIds,
-        ]),
-    ];
```

Comment:
Removing this logic because the table provider is not responsible for correcting the column order when the selected fields change. That's responsibility of the explore provider.